pkg/devspace/config/loader/validate.go

```diff
@@ -212,6 +212,12 @@ func validateHooks(config *latest.Config) error {
 		if hookConfig.Wait != nil && !hookConfig.Wait.Running && hookConfig.Wait.TerminatedWithCode == nil {
 			return errors.Errorf("hooks[%d].wait.running or hooks[%d].wait.terminatedWithCode is required if hooks[%d].wait is used", index, index, index)
 		}
+		if hookConfig.Container.ContainerName != "" && len(hookConfig.Container.LabelSelector) == 0 {
```

Comment:
You need to check for hookConfig.Container != nil here otherwise there might be a nil pointer